internal/hcl/modules/loader.go

```diff
@@ -72,38 +71,63 @@ func NewModuleLoader(path string, credentialsSource *CredentialsSource, logger *
 
 // downloadDir returns the path to the directory where remote modules are downloaded relative to the current working directory
 func (m *ModuleLoader) downloadDir() string {
-	return filepath.Join(m.Path, downloadDir)
+	return filepath.Join(m.cachePath, downloadDir)
 }
 
 // manifestFilePath is the path to the module manifest file relative to the current working directory
-func (m *ModuleLoader) manifestFilePath() string {
-	return filepath.Join(m.Path, manifestPath)
+func (m *ModuleLoader) manifestFilePath(projectPath string) string {
+	if m.cachePath == projectPath {
+		return filepath.Join(m.cachePath, manifestPath)
+	}
+
+	rel, _ := filepath.Rel(m.cachePath, projectPath)
+	sum := md5.Sum([]byte(rel)) //nolint
+	return filepath.Join(m.cachePath, ".infracost/terraform_modules/", fmt.Sprintf("%x.json", sum))
 }
 
-// tfManifestFilePath is the path to the terraform module manifest file relative to the current working directory.
-func (m *ModuleLoader) tfManifestFilePath() string {
-	return filepath.Join(m.Path, tfManifestPath)
+// tfManifestFilePath is the path to the Terraform module manifest file relative to the current working directory.
+func (m *ModuleLoader) tfManifestFilePath(path string) string {
+	return filepath.Join(path, tfManifestPath)
 }
 
 // Load loads the modules from the given path.
 // For each module it checks if the module has already been downloaded, by checking if iut exists in the manifest
 // If not then it downloads the module from the registry or from a remote source and updates the module manifest with the latest metadata.
-func (m *ModuleLoader) Load() (*Manifest, error) {
-	if m.newSpinner != nil {
-		spin := m.newSpinner("Downloading Terraform modules")
+func (m *ModuleLoader) Load(path string) (man *Manifest, err error) {
+	defer func() {
+		if man != nil {
+			man.cachePath = m.cachePath
+		}
+	}()
+
+	if m.NewSpinner != nil {
+		spin := m.NewSpinner("Downloading Terraform modules")
 		defer spin.Success()
 	}
 
 	manifest := &Manifest{}
-
-	_, err := os.Stat(m.manifestFilePath())
+	manifestFilePath := m.manifestFilePath(path)
+	_, err = os.Stat(manifestFilePath)
 	if errors.Is(err, os.ErrNotExist) {
 		m.logger.Debug("No existing module manifest file found")
 
-		_, err = os.Stat(m.tfManifestFilePath())
+		tfManifestFilePath := m.tfManifestFilePath(path)
+		_, err = os.Stat(tfManifestFilePath)
 		if err == nil {
-			manifest, err = readManifest(m.tfManifestFilePath())
+			manifest, err = readManifest(tfManifestFilePath)
 			if err == nil {
+				// let's make the module dirs relative to the path directory as later
+				// we'll look up the modules based on the cache path at the Infracost root (where the infracost.yml
+				// resides or where the --path autodetect started for multi-project)
+				for i, module := range manifest.Modules {
+					dir := path
+					if m.cachePath != "" {
+						dir, _ = filepath.Rel(m.cachePath, path)
+					}
+
+					manifest.Modules[i].Dir = filepath.Join(dir, module.Dir)
+				}
```

Comment:
I wasn't sure here if we should copy these to the `.infracost` directory instead. I've left that out for now. But I think we should probably migrate away from relying on the `.terraform` directory and instead do a conversion step when we first find it.